psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -1356,23 +1356,78 @@ std::vector<std::vector<std::tuple<double, int, int>>> Wavefunction::get_no_occu
     return no_occs;
 }
 
-double Wavefunction::get_variable(std::string label) {
-    std::string uc_label = label;
 
-    if (variables_.count(uc_label) == 0) {
-        throw PSIEXCEPTION("Wavefunction::get_variable: Requested variable " + label + " was not set!\n");
+int Wavefunction::has_scalar_variable(const std::string &key) {
+    std::string uc_key = key;
+    std::transform(uc_key.begin(), uc_key.end(), uc_key.begin(), ::toupper);
+    return variables_.count(uc_key);
+}
+
+int Wavefunction::has_array_variable(const std::string &key) {
+    std::string uc_key = key;
+    std::transform(uc_key.begin(), uc_key.end(), uc_key.begin(), ::toupper);
+    return arrays_.count(uc_key);
```

Comment:
@andysim ah, cast this to `bool` and change the return type? sounds good to me.

Reply:
If the return type is bool, the compiler will handle the cast for you.  It makes no difference, of course, but it means at first glance new programmers know that this is saying whether a var exists or not, rather than _how many_ exists (which makes no sense, of course).

Reply:
Or even just `return (arrays_.find(uc_key) != arrays_.end());`